cmd/ax/upgrade.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"archive/tar"
+	"bufio"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/google/go-github/github"
+)
+
+func getLatestReleaseData() *github.RepositoryRelease {
+	client := github.NewClient(nil)
+
+	release, _, err := client.Repositories.GetLatestRelease(context.Background(), "egnyte", "ax")
+	if err != nil {
+		fmt.Printf("Couldn't get a release information")
+		os.Exit(1)
+	}
+
+	return release
+}
+
+func getLatestAssetLink() string {
+	release := getLatestReleaseData()
+	var assetDownloadLink string
+
+	for _, element := range release.Assets {
+		assetLink := element.GetBrowserDownloadURL()
+		if strings.Contains(assetLink, runtime.GOOS) && strings.Contains(assetLink, runtime.GOARCH) {
+			assetDownloadLink = assetLink
+			break
+		}
+	}
+
+	return assetDownloadLink
+}
+
+func getLatestReleaseTag() string {
+	release := getLatestReleaseData()
+	return release.GetTagName()
+}
+
+func downloadFile(filepath string, url string) error {
+
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if _, err = io.Copy(out, resp.Body); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func upgradeVersion() error {
+	if latestVersion := getLatestReleaseTag(); version != latestVersion {
+		fmt.Printf("New version detected. Current version: %s. Latest version: %s\n", version, latestVersion)
+		inputScanner := bufio.NewScanner(os.Stdin)
+		fmt.Print("Do you want to upgrade an Ax to its latest version? (yes/no): ")
+		inputScanner.Scan()
+		text := inputScanner.Text()
+
+		if strings.ToLower(text) == "yes" || strings.ToLower(text) == "y" {
+			latestAssetURL := getLatestAssetLink()
+			downloadPath := "/tmp/egnyte_ax/ax.tar.gz"
+			extractionPath := "/tmp/egnyte_ax/"
```

Comment:
Should use [`TempFile`](https://golang.org/pkg/io/ioutil/#TempFile) and possibly [`TempDir`](https://golang.org/pkg/io/ioutil/#example_TempDir) for this.